AsyncRx/src/AsyncObserver.fs

```diff
@@ -86,3 +86,18 @@ module AsyncObserver =
             }
         }
 
+    let autoDetachObserver (obv: IAsyncObserver<'TSource>) : IAsyncObserver<'TSource>*(Async<IAsyncRxDisposable>->Async<IAsyncRxDisposable>) =
+        let disposables : ResizeArray<IAsyncRxDisposable> = ResizeArray ()
```

Comment:
Is it a problem that `ResizeArray` is not thread-safe?

Reply:
Yes, replaced with a mailbox agent.

Reply:
Looks good to me. Nice work.

Reply:
Thanks, let's try this